lean4-infoview/src/infoview/info.tsx

```diff
@@ -86,6 +86,7 @@ interface InfoDisplayProps extends InfoPinnable {
     termGoal?: InteractiveGoal;
     error?: string;
     rpcSess: RpcSession;
+    messagesRpcSess: RpcSession;
```

Comment:
> Why do messages have a separate RPC session? Is any given `InfoDisplay` not always ever "about" a single file?

Yes, the two session will always be for the same file (I believe).  What can differ though in practice is the session ID: since the messages and goals are updated on different schedules, the RpcRefs in the goal can be from a different session than the RpcRefs in the messages.  (For example if the worker is restarted and the messages are already updated but the not the goals.)  We don't want to send RpcRefs from one session to another session.  (An `RpcSession` object always has the same session ID, in case that wasn't clear.)

Reply:
I see. This is a bit awkward but important for performance in cases where the goal state is small but messages are huge or vice-versa, so not sure what to do about it.